ui2/src/pages/network/Clearing.tsx

```diff
@@ -0,0 +1,218 @@
+  services: Readonly<CreateEvent<Service, any, any>[]>;
+};
+
+export const ClearingServiceTable: React.FC<Props> = ({ services }) => {
+  const party = useParty();
+  const ledger = useLedger();
+
+  const offers = useStreamQueries(Offer).contracts;
+  const requests = useStreamQueries(Request).contracts;
+
+  const roles = useStreamQueries(Role).contracts;
+  const hasRole = roles.length > 0 && roles[0].payload.provider === party;
+
+  const terminateService = async (c: CreateEvent<Service>) => {
+    await ledger.exercise(Service.Terminate, c.contractId, { ctrl: party });
+  };
+
+  const [clearingAccountName, setClearingAccountName] = useState('');
+  const [marginAccountName, setMarginAccountName] = useState('');
+  const allocationAccountRules = useStreamQueries(AllocationAccountRule).contracts;
+  const allocationAccounts = allocationAccountRules
+    .filter(c => c.payload.account.owner === party)
+    .map(c => c.payload.account);
+  const allocationAccountNames: DropdownItemProps[] = allocationAccounts.map(a =>
+    createDropdownProp(a.id.label)
+  );
+
+  const assetSettlementRules = useStreamQueries(AssetSettlementRule).contracts;
+  const accounts = assetSettlementRules
+    .filter(c => c.payload.account.owner === party)
+    .map(c => c.payload.account);
+  const accountNames: DropdownItemProps[] = accounts.map(a => createDropdownProp(a.id.label));
+
+  const approveRequest = async (c: CreateEvent<Request>) => {
+    if (!hasRole) return; // TODO: Display error
+    console.log(roles[0].contractId);
+    await ledger.exercise(Role.ApproveClearingRequest, roles[0].contractId, {
+      clearingRequestCid: c.contractId,
+    });
+  };
+
+  const rejectRequest = async (c: CreateEvent<Request>) => {
+    if (!hasRole) return; // TODO: Display error
+    console.log(roles[0].contractId);
+    await ledger.exercise(Role.RejectClearingRequest, roles[0].contractId, {
+      clearingRequestCid: c.contractId,
+    });
+  };
+
+  const cancelRequest = async (c: CreateEvent<Request>) => {
+    await ledger.exercise(Request.Cancel, c.contractId, {});
+  };
+
+  const acceptOffer = async (c: CreateEvent<Offer>) => {
+    const clearingAccount = accounts.find(a => a.id.label === clearingAccountName);
+    const marginAccount = allocationAccounts.find(a => a.id.label === marginAccountName);
+    if (!clearingAccount || !marginAccount) return;
+    await ledger.exercise(Offer.Accept, c.contractId, { marginAccount, clearingAccount });
+  };
+
+  const withdrawOffer = async (c: CreateEvent<Offer>) => {
+    await ledger.exercise(Offer.Withdraw, c.contractId, {});
+  };
+
+  const rejectOffer = async (c: CreateEvent<Offer>) => {
+    await ledger.exercise(Offer.Decline, c.contractId, {});
+  };
+
+  return (
+    <div className="assets">
+      <Header as="h3">Current Services</Header>
+      <StripedTable
+        headings={['Service', 'Operator', 'Provider', 'Consumer', 'Role', 'Action' /* 'Details' */]}
+        rows={services.map((c, i) => {
```

Comment:
there is a loading prop for striped table now, if interested

Reply:
nice 👍